main.go

```diff
@@ -1,121 +1,38 @@
 package main
 
 import (
-	"encoding/json"
-	"io/ioutil"
+	"bufio"
+	"fmt"
 	"log"
 	"net/http"
-)
+	"os"
 
-const (
-	getPostsEndpoint    = "https://my-json-server.typicode.com/typicode/demo/posts"
-	getCommentsEndpoint = "https://my-json-server.typicode.com/typicode/demo/comments"
+	servicehandler "thien.com/service-handler"
 )
 
-type Post struct {
-	ID    int64  `json:"id"`
-	Title string `json:"title"`
-}
-type Comment struct {
-	ID     int64  `json:"id"`
-	Body   string `json:"body"`
-	PostID int64  `json:"postId"`
-}
-type PostWithCommentsResponse struct {
-	Posts []PostWithComments `json:"posts"`
-}
-type PostWithComments struct {
-	ID       int64     `json:"id"`
-	Title    string    `json:"string"`
-	Comments []Comment `json:"comments,omitempty"`
-}
-
 //TODO: how to separate API logic, business logic and response format logic
 func main() {
-	http.HandleFunc("/postWithComments", func(writer http.ResponseWriter, request *http.Request) {
-		// Get posts from api
-		posts, err := getPosts()
-		if err != nil {
-			log.Println("get posts failed with error: ", err)
-			writer.WriteHeader(500)
-			return
-		}
 
-		// Get comments from api
-		comments, err := getComments()
-		if err != nil {
-			log.Println("get comments failed with error: ", err)
-			writer.WriteHeader(500)
-			return
+	fmt.Print("\tPlease select render type:\n1: JSON\n2: XML\n\nInput: ")
+	scanner := bufio.NewScanner(os.Stdin)
+	var isJSON bool
+	for scanner.Scan() {
+		text := scanner.Text()
+		if text == "1" {
+			isJSON = true
+			break
 		}
-
-		// Combine and return response
-		postWithComments := combinePostWithComments(posts, comments)
-		resp := PostWithCommentsResponse{Posts: postWithComments}
-		buf, err := json.Marshal(resp)
-		if err != nil {
-			log.Println("unable to parse response: ", err)
-			writer.WriteHeader(500)
+		if text == "2" {
+			isJSON = false
+			break
```

Comment:
It's nice that you're adding this part, but the code is now tightly coupled with input is Stdin.
We can make the program accept any input by DI.